docs/proposals/d-no-change.md

```diff
@@ -0,0 +1,114 @@
+# Proposal D: No Changes
+
+The following proposal describes how existing OCI components can be used to provide features of reference types without introducing new manifest schemas or distribution APIs.
+
+## Artifact Definition
+
+The artifact is defined using the existing OCI Artifact definition (image-spec to a custom config media type):
+
+```jsonc
+{
+  "mediaType": "application/vnd.oci.image.manifest.v1+json",
+  "schemaVersion": 2,
+  "config": {
+    "mediaType": "<artifact-config-media-type>",
+    "size": 1234,
+    "digest": "sha256:a1a1a1..."
+  },
+  "layers": [
+    {
+      "mediaType": "<artifact-layer-media-type>",
+      "size": 1234,
+      "digest": "sha256:b2b2b2...",
+      "annotations": {
+        "com.example.metadata": "value"
+      }
+    }
+  ]
+}
+```
+
+Attaching an artifact to another object in the registry may be done with an OCI Index.
+For this example, the above artifact manifest is assumed to be `sha256:a0a0a0...`:
+
+```jsonc
+{
+  "mediaType": "application/vnd.oci.image.index.v1+json",
+  "schemaVersion": 2,
+  "manifests": [
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "digest": "sha256:010101...",
+      "size": 1234,
+      "platform": {
+        "architecture": "amd64",
+        "os": "linux"
+      }
+    },
+    { // the following is an artifact for the above linux/amd64 image
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "digest": "sha256:a0a0a0...",
+      "size": 1234,
+      "platform": {
+        "architecture": "amd64",
+        "os": "linux"
+      },
+      "annotations": {
+        "vnd.oci.artifact": "true", // this descriptor points to an artifact
+        "vnd.oci.artifact.type": "sig"
+      }
+    },
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "digest": "sha256:020202...",
+      "size": 1234,
+      "platform": {
+        "architecture": "arm64",
+        "os": "linux"
+      }
+    },
+    { // the following is an artifact for the above linux/arm64 image
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "digest": "sha256:b0b0b0...",
+      "size": 1234,
+      "platform": {
+        "architecture": "arm64",
+        "os": "linux"
+      },
```

Comment:
Is there a concern with this that a runtime might incorrectly pull this SBOM and interpret it as an image? It'd be looking for `mediaType == "image" && platform == $myplatform`, which this satisfies.

I guess that's what language introduced in https://github.com/opencontainers/image-spec/pull/880 is meant to guard against.

Reply:
There's certainly a concern of that. Also if we have a reference to an index, we could send up with an index pointing to an index, which may not be traversed by runtimes. Potentially we could use annotations rather than the platform section to associate one descriptor to another.